diff --git a/node_modules/@react-native-community/geolocation/android/src/main/java/com/reactnativecommunity/geolocation/PlayServicesLocationManager.java b/node_modules/@react-native-community/geolocation/android/src/main/java/com/reactnativecommunity/geolocation/PlayServicesLocationManager.java
index 423f7a5..04be4d3 100644
--- a/node_modules/@react-native-community/geolocation/android/src/main/java/com/reactnativecommunity/geolocation/PlayServicesLocationManager.java
+++ b/node_modules/@react-native-community/geolocation/android/src/main/java/com/reactnativecommunity/geolocation/PlayServicesLocationManager.java
@@ -167,14 +167,20 @@ public class PlayServicesLocationManager extends BaseLocationManager {
 
                 callbackHolder.success(location);
 
-                mFusedLocationClient.removeLocationUpdates(mSingleLocationCallback);
-                mSingleLocationCallback = null;
+                mFusedLocationClient.removeLocationUpdates(this);
+                if (mSingleLocationCallback == this) {
+                    mSingleLocationCallback = null;
+                }
             }
 
             @Override
             public void onLocationAvailability(@NonNull LocationAvailability locationAvailability) {
                 if (!locationAvailability.isLocationAvailable()) {
                     callbackHolder.error(PositionError.buildError(PositionError.POSITION_UNAVAILABLE, "Location not available (FusedLocationProvider/lastLocation)."));
+                    mFusedLocationClient.removeLocationUpdates(this);
+                    if (mSingleLocationCallback == this) {
+                        mSingleLocationCallback = null;
+                    }
                 }
             }
         };
